### ISSUE TITLE
Makefile does not appear to be rebuilding / regenerating gobin targets

### ISSUE THREAD

**Issue by chrislovecnm** [kubernetes/kops#3469]
When changes are made to our components that use gobin, we need to run it before we build dependencies.  Currently, kops and nodeup use gobin components.  The default make target should either figure out if we need the update, or run gobin.

/assign @alrs

**Comment by chrislovecnm** [comment]
I might be wrong, hopefully, I am, I am just not seeing the target being run.

``console
$ make
go install  -ldflags "-X k8s.io/kops.Version=1.8.0-alpha.1 -X k8s.io/kops.GitVersion=548a8291f " k8s.io/kops/cmd/kops/
```

We need to regenerate the gobin data stuff if we have made changes.  Are we?

**Comment by alrs** [comment]
@chrislovecnm That should be what is happening on lines 163 and 167 of the Makefile. @justinsb added this recently, so that when any of the files listed in `UPUP_MODELS_BINDATA_SOURCES` or `FEDERATION_MODELS_BINDATA_SOURCES` get modified the respective `bindata.go` files are rebuilt. If you want to verify that the `bindata.go` files are being rebuilt when needed you can manually delete them and run `make`, or you can can `make clean`, which should also delete them.

**Comment by alrs** [comment]
I also tested this path:

1. run `make clean`
2. run `make kops`. I saw `go-bindata`, `upup/models/bindata.go`,`federation/model/bindata.go` and `kops` get built.
3. run `make kops`. Nothing happens, because kops is already built.
4. add a newline to `federation/model/manifest.yaml`.
5. run `make kops`. `make` correctly doesn't bother rebuilding `go-bindata` or `upup/models/bindata.go`, but it does see that `federation/model/bindata.go` has changed, so it rebuilds that and `kops`, since `kops` depends on `federation/model/bindata.go`.
